trie/hexary.py

```diff
@@ -131,20 +154,58 @@ def _get(self, root_hash, trie_key):
         else:
             raise Exception("Invariant: This shouldn't ever happen")
 
-    def _traverse(self, root_hash, trie_key) -> Tuple['Node', Tuple[int, ...]]:
+    def traverse(self, trie_key_input: Nibbles) -> HexaryTrieNode:
```

Comment:
Not clear to me what `trie_key_input` is.  Also it is typed as `Nibbles` but then the first thing this function does is cast this value to `Nibbles(trie_key_input)`.  I'm not understanding what or why that is happening.

Reply:
`Nibbles` is just a nice way to validate that each int in the tuple is in `range(16)`, if someone passes in just a tuple of ints (since type hints are just a hint, not formally required). I'll add a comment about it being for validation.